core/kernel/dt.c

```diff
@@ -125,6 +125,39 @@ int dt_map_dev(const void *fdt, int offs, vaddr_t *base, size_t *size)
 	return 0;
 }
 
+int dt_map_secure_dev(const void *fdt, int offs, vaddr_t *base, size_t *size)
+{
+	paddr_t pbase = 0;
+	vaddr_t vbase = 0;
+	size_t sz = 0;
+	int st = 0;
+
+	assert(cpu_mmu_enabled());
+
+	st = _fdt_get_status(fdt, offs);
+	if (st == DT_STATUS_DISABLED)
+		return -1;
+
+	pbase = _fdt_reg_base_address(fdt, offs);
+	if (pbase == DT_INFO_INVALID_REG)
+		return -1;
+	sz = _fdt_reg_size(fdt, offs);
+	if (sz == DT_INFO_INVALID_REG_SIZE)
+		return -1;
```

Comment:
The err_code(-1) returned by different exception branches. Maybe inconvenient for debug.

Reply:
Right. This is legacy and could move to `TEE_Result` return value.